src/libsyntax/parse/parser.rs

```diff
@@ -4265,11 +4287,27 @@ impl<'a> Parser<'a> {
         maybe_whole!(self, NtGenerics);
 
         if self.eat(&token::Lt) {
-            let lifetime_defs = self.parse_lifetime_defs()?;
+            // When encounter attributes in generics list, do not yet
+            // know if it is attached to lifetime or to type param.
```

Comment:
Nit: slightly iffy grammar, probably s/encounter/we encounter and s/do not/we do not